### PR TITLE
fix sandbox regressions

### PR DESCRIPTION
**Pull request by kevodwyer**
- handle path when app is launchable
- re-calculate thumbnail only if file is writable
- better handling of read only access

### CODE REVIEW

File: assets/apps/sandbox/sandbox.js [hunk 1]
```diff
@@ -72,6 +72,9 @@ function load(appName, appPath, allowBrowsing, theme, chatId, username, props) {
             path = path.length > 0 ? path + '&theme=' + theme : '?theme=' + theme;
             path = chatId.length > 0 ? path + '&chatId=' + chatId : path;
             path = path + '&username=' + username;
+            if (props.isFileWritable == true) {
+                path = path + '&isFileWritable=' + props.isFileWritable;
```
Comment (ianopolous) [inline assets/apps/sandbox/sandbox.js:76]: What if it opens a dir? Can we keep it general by calling it isPathWritable?

Comment (kevodwyer) [reply assets/apps/sandbox/sandbox.js:76]: isFileWritable is specific. This property will not be set if app is a folder action. We currently do not have any apps that are folder actions (beyond my testing apps), i don't want to over-design until i have  a use case to confirm with.

Comment (ianopolous) [reply assets/apps/sandbox/sandbox.js:76]: Sure but if people write apps based on this and we change it they will break. It is talking about the path supplied being writable correct?

Comment (kevodwyer) [reply assets/apps/sandbox/sandbox.js:76]: It is only a convienence so a file app can conditionally show a save button.
I have no problem adding a isPathWritable for folders down the line, but i think that will require more thought, ie multiple folders.
isFileWritable has a specific meaning for a specific case.

Comment (ianopolous) [reply assets/apps/sandbox/sandbox.js:76]: What is already set in stone is that we have a path arg that is provided to the app right? All we are doing is providing metadata about that path. It's doesn't matter if it's a file or folder.

Comment (kevodwyer) [reply assets/apps/sandbox/sandbox.js:76]: I will change isFileWritable to isPathWritable

**Review by ianopolous — APPROVED** [review]

**Pull request merged**